dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -128,22 +128,27 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
-    PushNotificationRegistrationToken PushNotificationManager::RegisterActivator(PushNotificationActivationInfo const& details)
+    void PushNotificationManager::RegisterActivator(PushNotificationActivationInfo const& details)
     {
         THROW_HR_IF_NULL(E_INVALIDARG, details);
 
         GUID taskClsid = details.TaskClsid();
         THROW_HR_IF(E_INVALIDARG, taskClsid == GUID_NULL);
 
-        auto registrationOptions = details.Options();
-        THROW_HR_IF(E_INVALIDARG, WI_AreAllFlagsClear(registrationOptions, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator));
+        auto registrationOptions = details.Activators();
+        THROW_HR_IF(E_INVALIDARG, WI_AreAllFlagsClear(registrationOptions, PushNotificationRegistrationActivators::PushTrigger | PushNotificationRegistrationActivators::ComActivator));
 
         DWORD cookie = 0;
         IBackgroundTaskRegistration registeredTask = nullptr;
         BackgroundTaskBuilder builder = nullptr;
 
-        if (WI_IsFlagSet(registrationOptions, PushNotificationRegistrationOptions::PushTrigger))
+        if (WI_IsFlagSet(registrationOptions, PushNotificationRegistrationActivators::PushTrigger))
         {
+            {
+                auto lock = s_lock.lock_exclusive();
+                THROW_HR_IF(E_INVALIDARG, s_taskRegistration);
```

Comment:
Is it intentional to throw if the developer calls RegisterActivator twice for the same registrationOption?